Tasks/InstallSSHKey/installsshkey-util.ts

```diff
@@ -0,0 +1,118 @@
+
+    constructor() {
+    }
+
+    private getExecutable(executable: string):string {
+        let isWindows: RegExpMatchArray = os.type().match(/^Win/);
+        if (isWindows && this.baseDir) {
+            executable = path.join(this.baseDir, path.join(this.sshGitExternalsDir, executable));
+            executable += '.exe';
+        }
+        return executable;
+    }
+
+    public runAgent() {
+        // Expected output sample:
+        // SSH_AUTH_SOCK=/tmp/ssh-XVblDhTvcbC3/agent.24196; export SSH_AUTH_SOCK;
+        // SSH_AGENT_PID=4644; export SSH_AGENT_PID; echo Agent pid 4644;
+        let agentResults: trm.IExecSyncResult = tl.execSync(this.getExecutable('ssh-agent'), null);
+
+        let elements: string[] = agentResults.stdout.split(';');
+        for (let i:number = 0; i < elements.length; ++i) {
+            let keyValue : string[] = elements[i].split('=');
+            if (keyValue && keyValue.length >= 2) {
+                let key: string = keyValue[0].trim();
+                let value: string =  keyValue[1].trim();
+                tl.debug('Key=' + key + ' value=' + value);
+                if (sshAgentPidEnvVariableKey === key) {
+                    tl.setVariable(key, value);
+                    tl.setTaskVariable(postKillAgentSetting, value);    
+                }
+                else if (sshAgentSockEnvVariableKey === key) {
+                    tl.setVariable(key, value);
+                }
+            } else {
+                tl.debug('Skipping ' + elements[i]);
+            }
+        }
+    }
+
+    public installKey(publicKey: string, privateKeyLocation: string) {
+        tl.debug('Get a list of the SSH keys in the agent');
+        let results: trm.IExecSyncResult = tl.execSync(this.getExecutable('ssh-add'), '-L');
+
+        let publicKeyComponents:string[] = publicKey.split(' ');
+        if (publicKeyComponents.length <= 1) {
+            throw tl.loc('SSHPublicKeyMalformed');
+        }
+
+        let publicKeyHash: string = publicKeyComponents[1];
+        tl.debug('Checking for public SSH key: ' + publicKeyHash);
+        if (results.stdout.indexOf(publicKeyHash) !== -1) {
+            throw tl.loc('SSHKeyAlreadyInstalled');
+        }
+
+        tl.debug('Adding the SSH key to the agent ' + privateKeyLocation);
+        results = tl.execSync(this.getExecutable('ssh-add'), privateKeyLocation);
+        if (results.error) {
+            throw tl.loc('SSHKeyInstallFailed');
+        }
+        tl.setTaskVariable(postDeleteKeySetting, privateKeyLocation);
+
+        results = tl.execSync(this.getExecutable('ssh-add'), null);
+    }
+
+    public deleteKey(key: string) {
+        let deleteKey: string = tl.getTaskVariable(postDeleteKeySetting);
+        if (deleteKey) {
+            tl.debug('Deleting Key: ' + deleteKey);
+            tl.execSync(path.join(external, 'ssh-add.exe'), '-d ' + deleteKey);
```

Comment:
There is a risk of this not working when there are spaces in the `deleteKey` path.  If you use a ToolRunner from the task library and add each item as a formal parameter ('-d', deleteKey), I think it will take care of putting quotes around the path for you if it has spaces.